11/Zaubeerer/test_generators.py

```diff
@@ -0,0 +1,57 @@
+import generators
+from collections import Counter
+from itertools import islice
+
+FILE_PATTERN = '*.py'
+LINE_PATTERN = "import "
+
+def test_gen_files():
+    expected_files = ["test_generators.py", "generators.py"]
+
+    assert list(generators.gen_files(FILE_PATTERN)) == expected_files
+
+def test_gen_lines():
+    expected_lines = ['import generators\n',
+                      'from collections import Counter\n', 
+                      'from itertools import islice\n',
+                      '\n', 
+                      "FILE_PATTERN = '*.py'\n", 
+                      'LINE_PATTERN = "import "\n', 
+                      ]
+
+    test_files = ["test_generators.py"]
+
+    assert list(generators.gen_lines(test_files))[:6] == expected_lines
+
+
+def test_gen_grep():
+    lines = ['import generators\n', 
+             '\n', 
+             "FILE_PATTERN = '*.py'\n", 
+             'LINE_PATTERN = "import "\n', 
+             '\n',
+             ]
+
+    expected_lines = ["generators"]
+
+    assert list(generators.gen_grep(lines, LINE_PATTERN)) == expected_lines
+
+def test_gen_count():
+
+    lines = ["generators", "numpy", "generators"]
+
+    expected_counts = [Counter({'generators': 2, 'numpy': 1})]
+
+    assert list(generators.gen_count(lines)) == expected_counts
+
+def test_integrated():
+
+    expected_counter = Counter({'re': 8, 'glob': 6, 'collections': 2, 'sys': 1, 'glob, os': 1, 'generators': 1, 'ast': 1})
+
+    files = generators.gen_files("../*/*.py")
```

Comment:
this did not work on my system, can you make the path a constant / load from env?

Reply:
What do you mean?

I used `../*/*py` as described in the challenge:

`$ for i in ../*/*py; do grep ^import $i|sed 's/import //g' ; done | sort | uniq -c | sort -nr`

Reply:
yeah ok on us, ok here, in real life put it in config

Reply:
Alright. :)

config then refers to an external config file or a variable / constant at the file head?